crates/cheatcodes/spec/src/vm.rs

```diff
@@ -286,6 +306,22 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Unsafe)]
     function loadAllocs(string calldata pathToAllocsJson) external;
 
+    // -------- Record Debug Traces --------
+
+    /// Records the debug trace during the run.
+    #[cheatcode(group = Evm, safety = Safe)]
+    function startDebugTraceRecording() external;
+
+    /// Stop debug trace recording and returns the total size of the recorded debug trace.
+    #[cheatcode(group = Evm, safety = Safe)]
+    function stopDebugTraceRecording() external returns (uint256 size);
+
+
+    /// Get the recorded debug trace by step index.
+    #[cheatcode(group = Evm, safety = Safe)]
+    function getDebugTraceByIndex(uint256 index) external returns (DebugStep memory step);
```

Comment:
did you experience OOMs if returning an array of steps from `stopDebugTraceRecording`? I'd expect it to be fine given that we use the same recording in debugger

Reply:
yeah, that was originally why I made the change to use get-by-index style instead. 

However, that end up helping to certain degree, but not solving the issue fully, and I end up realize the issue was that I forgot to turn the tracer config back before.

I did a final check today, and the get-by-index style does have some gas improvement; but yeah, if I removed the line that turns the tracer config back on stop recording, the forge will OOM (and crash, not just EVM returning OOM)

With the current setup (having tracer config being set back), the gas cost is around 2x when returning all debug traces as the checker tool will pass around the debug traces in internal functions. I need to set the `gas_limit = "18446744073709551615"` in both way anyway as both styles are exceeding default gas limit and will cause EVM OOM (not crashing). Given that, I agree it makes sense to revert the style back to the one returning all traces.